mealie/routes/groups/controller_cookbooks.py

```diff
@@ -51,27 +56,39 @@ def get_all(self, q: PaginationQuery = Depends(PaginationQuery)):
         return response
 
     @router.post("", response_model=ReadCookBook, status_code=201)
-    def create_one(self, data: CreateCookBook):
+    def create_one(self, request: Request, data: CreateCookBook):
         data = mapper.cast(data, SaveCookBook, group_id=self.group_id)
-        val = self.mixins.create_one(data)
+        cookbook = self.mixins.create_one(data)
 
-        if val:
+        if cookbook:
             self.event_bus.dispatch(
-                self.user.group_id,
-                EventTypes.cookbook_created,
-                msg=self.t("notifications.generic-created", name=val.name),
-                event_source=EventSource(event_type="create", item_type="cookbook", item_id=val.id, slug=val.slug),
+                request=request,
+                group_id=self.user.group_id,
+                event_type=EventTypes.cookbook_created,
+                document_data=EventCookbookData(operation=EventOperation.create, cookbook_id=cookbook.id),
+                message=self.t("notifications.generic-created", name=cookbook.name),
             )
```

Comment:
This seems like a really verbose way to call this in each instance. I think there's going to be a ton of these in the code-base. Is there any way to abstract some of the common parameters? I have two ideas

1. Utilize the controllers to instantiate the EventBuss and have a `publish` event method that pre-captures the Request and group_id so we don't have to pass those in.
2. Have a class method on the EventBusService that requires a the group_id and request method be provided and then have a special method on the event bus that wraps `dispatch` but pre-sets the request and group_id.

Reply:
That makes sense, would that go in the router? i.e. the router captures the request/user and sends it to the event bus. That said I think the only thing I needed the request for was the token which has the integration id, so we may not event need the request if we're moving that to a custom header.

Actually, now that I think about it, isn't the controller instantiated with the group_id? Then we could just pass it to the event bus.

Reply:
Yeah, the controller already has most everything you need. You could probably keep the JWT implementation but simplify it a bit by having a `integration_id` property on the controller that would return the `integration_id` from the JWT, and then have a method on the controller that contains those values to simplify the call in each route.

I don't think there is too many valid cases to reach for the router with the Controller based system we have. That sort of fills the void there.

Reply:
This is probably a stupid question, but how do I get the token/integration id to the controller? I see `BaseUserController` which has a `PrivateUser` that depends on `get_current_user` which gets the user from the token, but I don't actually see how the token gets passed to `get_current_user`. Looks like some FastAPI dependency magic.